### PR TITLE
docs: Update and more general wording for "docker container"

### PR DESCRIPTION
**Pull request by dklimpel**
##### Checklist

- [x] `make test-all` (UNIX) passes. CI will also test this
- [ ] unit and/or integration tests are included (if applicable)
- [x] documentation is changed or added (if applicable)

### Description of change

Fixes: #931 

----
📚 Documentation preview 📚: https://goss--934.org.readthedocs.build/en/934/

### CODE REVIEW

**Comment by aelsabbahy** [comment]
Thank you for submitting this.

Couple of feedbacks on this:

1. Removing docker from kgoss and in a few other places makes sense. But this PR is a little aggressive and removes docker from everything.  dgoss is a docker (and also podman) wrapper for goss, dcgoss is a docker-compose wrapper for Goss 

2. CI is failing, not sure why yet.

File: README.md [hunk 1]
```diff
@@ -77,7 +77,7 @@ VERSION=v0.3.10
 curl -L "https://github.com/goss-org/goss/releases/download/${VERSION}/goss-linux-amd64" -o /usr/local/bin/goss
 chmod +rx /usr/local/bin/goss
 
-# (optional) dgoss docker wrapper (use 'master' for latest version)
+# (optional) dgoss container wrapper (use 'master' for latest version)
```
Comment (dklimpel) [inline README.md:80]:
```suggestion
# (optional) dgoss docker wrapper (use 'master' for latest version)
```

**Review by dklimpel — COMMENTED** [review]
> 2\. CI is failing, not sure why yet.

The CI has problems with `goss_int_test_alpine3_386`. I think the CI is not reliable. Since the last successful run (https://github.com/goss-org/goss/commit/a6451d2f8b7af34f56beafd43bb81091fb10e2ab) there were only changes at documentation.

I will revert one change. Feel free to revert more or mark the positions.
IMO docker and podman are wrapper for container engines. And both runs container. There are no special docker container.

**Review by petemounce — APPROVED** [review]

File: extras/dcgoss/README.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ containers. It is based on `dgoss`.
```
Comment (aelsabbahy) [inline extras/dcgoss/README.md:9]: lets revert the changes in this file, since it's dcgoss aka docker-compose goss.

File: README.md [hunk 3]
```diff
@@ -14,11 +14,11 @@
 <a href="https://asciinema.org/a/4suhr8p42qcn6r7crfzt6cc3e?autoplay=1" target="_blank"><img src="https://cloud.githubusercontent.com/assets/6783261/17330426/ce7ad066-5894-11e6-84ea-29fd4207af58.gif" alt="asciicast"></a>
 <!-- markdownlint-enable line-length no-inline-html -->
 
-**Note:** For testing docker containers see the [dgoss](https://github.com/goss-org/goss/tree/master/extras/dgoss) wrapper.
+**Note:** For testing containers see the [dgoss](https://github.com/goss-org/goss/tree/master/extras/dgoss) wrapper.
 Also, user submitted wrapper scripts for Kubernetes [kgoss](https://github.com/goss-org/goss/tree/master/extras/kgoss)
 and Docker Compose [dcgoss](https://github.com/goss-org/goss/tree/master/extras/dcgoss).
 
-**Note:** For some Docker/Kubernetes healthcheck, health endpoint, and
```
Comment (aelsabbahy) [inline README.md:21]: The blog post references docker, this line should be reverted.

**Review by aelsabbahy — COMMENTED** [review]
small changes, otherwise looks good.

File: extras/dcgoss/README.md (hunk 2 above)
Comment (dklimpel) [reply extras/dcgoss/README.md]: Done.

File: README.md (hunk 3 above)
Comment (dklimpel) [reply README.md]: Done.
